src/libraries/System.Text.Encoding.CodePages/src/Data/Tools/EncodingDataGenerator.cs

```diff
@@ -0,0 +1,466 @@
+                    continue;
+                }
+
+                string name = line.Value[1].Trim().ToLowerInvariant();
+                if (name != line.Value[1])
+                {
+                    Console.WriteLine("Code page name in file {0} at line {1} has whitespace or upper-case characters.  Was: ->{2}<-, Using ->{3}<-", path, line.Key, line.Value[1], name);
+                }
+
+                string englishName = line.Value[2].Trim();
+                if (englishName != line.Value[2])
+                {
+                    Console.WriteLine("English name in file {0} at line {1} has whitespace.  Was: ->{2}<-, Using ->{3}<-", path, line.Key, line.Value[2], englishName);
+                }
+
+                KeyValuePair<string, string> names = KeyValuePair.Create(name, englishName);
+
+                KeyValuePair<string, string> existing;
+                if (preferredNames.TryGetValue(codepage, out existing))
+                {
+                    if (names.Equals(existing))
+                    {
+                        Console.WriteLine("Code page names {0} for code page {1} in file {2} at line {3} is a duplicate entry, and can be removed.", names, codepage, path, line.Key);
+                    }
+                    else
+                    {
+                        Console.WriteLine("Code page {0} in file {1} at line {2} is mapped to multiple names; new is {3}, old was {4}", codepage, path, line.Key, names, existing);
+                    }
+                }
+                else
+                {
+                    preferredNames[codepage] = names;
+                }
+            }
+
+            return preferredNames;
+        }
+
+        private IEnumerable<KeyValuePair<int, string[]>> DelimitedFileRows(string path, int columns = 0)
+        {
+            using (var input = File.OpenText(path))
+            {
+                int lineNumber = 1;
+                string line;
+
+                for (; (line = input.ReadLine()) != null; ++lineNumber)
+                {
+                    if (line.StartsWith(CommentIndicator) || string.IsNullOrWhiteSpace(line))
+                    {
+                        continue;
+                    }
+
+                    string[] values = line.Split(FieldDelimiter);
+
+                    if (columns > 0 && values.Length != columns)
+                    {
+                        Console.WriteLine("Parsing mapping in file {0}, line {1}.  Expected {2} fields, saw {3}: {4}", path, lineNumber, columns, values.Length, line);
+                    }
+
+                    yield return KeyValuePair.Create(lineNumber, line.Split(FieldDelimiter));
+                }
+            }
+        }
+
+        private const string Header =
+
+@"// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
```

Comment:
```suggestion

```

We're about to enable failing the build if a file lacks the appropriate license header, and with this being a contiguous comment, the tool will see it as not being the right header.

Reply:
I think the GitHub suggestion feature does not allow a single line to be replaced by empty line (workaround is to to select one line before and after). In this case, we want an empty line between header and a non-header comment to satisfy IDE license header rule.

Reply:
I'll fix this manually. Thanks @am11